Src/UnitTests/CoravelUnitTests/Queuing/CancellableInvocableForQueueTests.cs

```diff
@@ -82,9 +101,37 @@ public Task Invoke()
                 {
                     Interlocked.Increment(ref TokensCancelled);
                 }
+                return Task.CompletedTask;
+            }
+        }
+        
+        private class TestCancellableInvocableWithPayload : IInvocable, IInvocableWithPayload<TestPayload>, ICancellableTask
+        {
+            /// <summary>
+            /// Static fields keeps track of all cancelled tokens count.
+            /// </summary>
+            public static int TokensCancelled = 0;
+            public TestPayload Payload { get; set; }
+            public TestCancellableInvocableWithPayload() {}
+
+            public CancellationToken Token { get; set; }
+
+            public Task Invoke()
+            {
+                if(this.Token.IsCancellationRequested)
+                {
+                    Interlocked.Increment(ref TokensCancelled);
+                }
+                Thread.Sleep(10000);
```

Comment:
Could we remove this `Thread.Sleep()`. It should work without it I believe (just like the other tests)